File: src/main/java/com/kamikazejam/datastore/base/field/FieldProvider.java
package com.kamikazejam.datastore.base.field;

import org.jetbrains.annotations.NotNull;

/**
 * Represents any object that can provide a FieldWrapper.
 * This is used to allow both FieldWrapper instances and wrapper collection classes
 * to be used in Store.getCustomFields().
 */
public interface FieldProvider {
    /**
     * Gets the underlying FieldWrapper that this object provides or represents.
     * @return The FieldWrapper instance
     */
    @NotNull FieldWrapper<?> getFieldWrapper();
}

File: src/main/kotlin/com/kamikazejam/datastore/base/field/FieldProvider.kt
package com.kamikazejam.datastore.base.field

import com.kamikazejam.datastore.base.Store

/**
 * Represents any object that can provide a [FieldWrapper].
 *
 * This is used to allow all implementations like [OptionalField], [RequiredField], etc.
 * to be returned in [Store.getCustomFields]
 */
interface FieldProvider {
    /**
     * Gets the underlying FieldWrapper that this object provides or represents.
     * @return The FieldWrapper instance
     */
    val fieldWrapper: FieldWrapper<*,*>
}